src/cfg/structured.rs

```diff
@@ -91,3 +105,131 @@ impl StructuredBlock {
         }
     }
 }
+
+impl StructuredFunction {
+    pub fn to_function(&self) -> Function {
+        let mut builder = StructuredCfgBuilder {
+            code: vec![],
+            scopes: vec![],
+            name_counter: 0,
+        };
+        self.block.to_code(&mut builder);
+
+        Function {
+            name: self.name.clone(),
+            args: self.args.clone(),
+            instrs: builder.code,
+            pos: None,
+            return_type: None,
+        }
+    }
+}
+
+pub struct StructuredCfgBuilder {
+    code: Vec<Code>,
+    scopes: Vec<String>,
+    name_counter: usize,
+}
+
+impl StructuredCfgBuilder {
+    fn fresh(&mut self) -> String {
+        self.name_counter += 1;
+        format!("sblock___{}", self.name_counter - 1)
+    }
+
+    fn scope_break_to(&self, num: i64) -> String {
```

Comment:
what is `num` an why must it be greater than 0? Can you add doc comments?

Reply:
Added a comment (: